src/Palladium/Entity/Identity.php

```diff
@@ -165,7 +166,11 @@ public function setToken($token)
 
     public function generateToken()
     {
-        $this->token = bin2hex(random_bytes(Identity::TOKEN_SIZE));
+        try {
+            $this->token = bin2hex(random_bytes(Identity::TOKEN_SIZE));
+        } catch (\Exception $e) {
+            throw new FailGenerateToken;
```

Comment:
You should call it `TokenGenerationFailed`, to match the current (though - loose) naming convention. And probably extend `RuntimeException`, since you basically can only get this if `/dev/urandom` has been hidden.

Reply:
agree.